core-library/src/main/java/org/silverpeas/core/admin/user/dao/UserSearchCriteriaForDAO.java

```diff
@@ -127,159 +134,125 @@ public UserSearchCriteriaForDAO onDomainId(String domainId) {
     // all users that are part of the specified domain or that have administration priviledges
     // (the administrators should be visible by anyone in order to be contacted)
     if (isDefined(domainId)) {
-      tables.add("st_user");
-      getFixedQuery().append("st_user.domainId = ").append(Integer.valueOf(domainId));
+      criteria.onDomainId(domainId);
     }
     return this;
   }
 
   @Override
-  public SearchCriteria onUserIds(String... userIds) {
+  public UserSearchCriteriaForDAO onUserIds(String... userIds) {
     if (userIds != ANY) {
-      tables.add("st_user");
-      StringBuilder[] sqlLists = asSQLList(userIds);
-      StringBuilder theQuery = getFixedQuery().append("(st_user.id in ").append(sqlLists[0]);
-      for (int i = 1; i < sqlLists.length; i++) {
-        theQuery.append(" or st_user.id in ").append(sqlLists[i]);
-      }
-      theQuery.append(")");
+      criteria.onUserIds(userIds);
     }
     return this;
   }
 
   @Override
   public UserSearchCriteriaForDAO onUserStatesToExclude(final UserState... userStates) {
-    Collections.addAll(userStatesToExclude, userStates);
-    String tmpFilter = filter.toString().replaceFirst(" (and|or) $", "");
-    filter = new StringBuilder(tmpFilter);
+    criteria.onUserStatesToExclude(userStates);
     return this;
   }
 
-  public String toSQLQuery(String fields) {
-    String ordering = ORDER_BY_LASTNAME;
-    if (fields.toLowerCase().matches("(count|max|min)\\(.*\\)")) {
-      ordering = "";
+  JdbcSqlQuery toSQLQuery(String fields) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "toSQLQuery" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)